### PR TITLE
Ref #151 - adds ability to set default base URL

### PR DESCRIPTION
**Pull request by jacobdfriedmann**

### CODE REVIEW

**Comment by jacobdfriedmann** [comment]
Creates the ability to specify a default base URL.

```javascript
APIActionCreator.setDefaultBaseURL('http://myservice.com/');
```

```javascript
var aac = new APIActionCreator({
    displayName: 'api',
    doThing: {
        // Request made to http://myservice.com/test
        route: '/test',
        method: 'GET'
    },
    doOtherThing: {
        // Override default with fully qualified URL
        // request made to http://myotherservice.com/test
        route: 'http://myotherservice.com/test',
        method: 'GET'
    }
});
```

File: src/APIActionCreator.es6.js [hunk 1]
```diff
@@ -181,9 +207,19 @@ export default class APIActionCreator extends ActionCreator {
 				request.headers = Object.assign({}, defaultHeaders.headers || {}, request.headers || {});
 			}
 
+			// If the user has defined a default base URL, append the route to the
+			// base URL if, and only if, the route itself is not a fully qualified
+			// URL.
+			let requestRoute = request.route ? request.route : route;
+			if (defaultBaseURL.url && requestRoute.indexOf('http') !== 0 && requestRoute.indexOf('//') !== 0) {
```
Comment (jhorwit2) [inline src/APIActionCreator.es6.js]: please use `.startsWith`

File: src/APIActionCreator.es6.js [hunk 2]
```diff
@@ -181,9 +207,19 @@ export default class APIActionCreator extends ActionCreator {
 				request.headers = Object.assign({}, defaultHeaders.headers || {}, request.headers || {});
 			}
 
+			// If the user has defined a default base URL, append the route to the
+			// base URL if, and only if, the route itself is not a fully qualified
+			// URL.
+			let requestRoute = request.route ? request.route : route;
```
Comment (jhorwit2) [inline src/APIActionCreator.es6.js]: `route` will always be defined. It's a precondition held up by an invariant on LOC 158

File: test/src/api-action-creator-tests.js [hunk 3]
```diff
@@ -614,6 +633,108 @@ describe('APIActionCreators', function () {
                 APIActionCreator.setDefaultHeaders(true);
             }).should.throw();
         });
+
+    });
+
+	describe('when setting default base URL', function () {
+        beforeEach(function () {
```
Comment (jhorwit2) [inline test/src/api-action-creator-tests.js]: spacing is off

File: src/APIActionCreator.es6.js [hunk 4]
```diff
@@ -23,6 +23,9 @@ const ActionCreator = require('./ActionCreator.es6');
 let defaultHeaders = {
 	headers: undefined // no headers by default
 };
+let defaultBaseURL = {
```
Comment (jhorwit2) [inline src/APIActionCreator.es6.js]: why is this an object?

Oh, I guess bc default headers? Idk why that is either. Would you mind making a `defaultOptions` object w/ headers & url? Should be cleaner and less stuttering when reading.

File: src/APIActionCreator.es6.js (hunk 2 above)
Comment (jacobdfriedmann) [inline src/APIActionCreator.es6.js]: `route` will be defined, but `request.route` may or may not be defined, depending on whether or not is added in the `createRequest` implementation. Based on the code that was here before `request.route` is suppose to override `route` if defined, hence this line.

Comment (jhorwit2) [inline src/APIActionCreator.es6.js]: I'll fix this in another diff. route should never be a thing in createRequest (as per docs)

**Pull request merged**